crates/l2/prover/src/prover_client.rs

```diff
@@ -24,88 +24,96 @@ struct ProverClient {
 }
 
 impl ProverClient {
-    pub fn new(prover_server_endpoint: String) -> Self {
+    pub fn new(config: ProverClientConfig) -> Self {
         Self {
-            prover_server_endpoint,
+            prover_server_endpoint: config.prover_server_endpoint,
         }
     }
 
     pub async fn start(&self) {
         let mut prover = Prover::new();
 
         loop {
-            match self.request_new_data() {
-                Ok((Some(block_number), input)) => {
+            match self.request_new_input() {
+                Ok((block_number, input)) => {
                     match prover.set_input(input).prove() {
                         Ok(proof) => {
-                            if let Err(e) = self.submit_proof(block_number, proof) {
-                                // TODO: Retry
+                            if let Err(e) =
+                                self.submit_proof(block_number, proof, prover.id.to_vec())
+                            {
+                                // TODO: Retry?
                                 warn!("Failed to submit proof: {e}");
                             }
                         }
                         Err(e) => error!(e),
                     };
                 }
-                Ok((None, _)) => sleep(Duration::from_secs(10)).await,
                 Err(e) => {
-                    sleep(Duration::from_secs(10)).await;
+                    sleep(Duration::from_secs(5)).await;
```

Comment:
File an issue to receive this duration as config.

Reply:
added in this PR so we don't forget